### PR TITLE
chore(datalist): reduce mobile borders to 2px

### LINKED ISSUE #6555: DataList - decrease border for mobile version
**Describe the feature**
Per @kaylachumley: Change the border width on mobile to be border-width: 2px (global/border/width/200)? The 8px border is bit thick. The default border color is fine. especially since it's not as thick anymore. Everything else looks good.

### PR DESCRIPTION
**Pull request by evwilkin**
Closes #6555 

This PR updates the default DataList and DataListItem borders from `--pf-t--global--spacer--sm` to `--pf-t--global--border--width--strong`

### CODE REVIEW

File: src/patternfly/components/DataList/data-list.scss [hunk 1]
```diff
@@ -5,7 +5,7 @@
   --#{$data-list}--FontSize: var( --pf-t--global--font--size--body);
   --#{$data-list}--LineHeight: var(--pf-t--global--font--line-height--body);
   --#{$data-list}--BorderBlockStartColor: var(--pf-t--global--border--color--default);
-  --#{$data-list}--BorderBlockStartWidth: var(--pf-t--global--spacer--sm);
+  --#{$data-list}--BorderBlockStartWidth: var(--pf-t--global--border--width--200);
```
Comment (mcoker) [inline src/patternfly/components/DataList/data-list.scss:8]:
```suggestion
  --#{$data-list}--BorderBlockStartWidth: var(--pf-t--global--border--width--strong);
```

File: src/patternfly/components/DataList/data-list.scss [hunk 2]
```diff
@@ -21,7 +21,7 @@
   --#{$data-list}__item--m-selected--BackgroundColor: var(--pf-t--global--background--color--primary--clicked);
   --#{$data-list}__item--m-clickable--OutlineOffset: calc(-1 * var(--pf-t--global--spacer--xs));
   --#{$data-list}__item--BorderBlockEndColor: var(--pf-t--global--border--color--default);
-  --#{$data-list}__item--BorderBlockEndWidth: var(--pf-t--global--spacer--sm);
+  --#{$data-list}__item--BorderBlockEndWidth: var(--pf-t--global--border--width--200);
```
Comment (mcoker) [inline src/patternfly/components/DataList/data-list.scss:24]:
```suggestion
  --#{$data-list}__item--BorderBlockEndWidth: var(--pf-t--global--border--width--strong);
```

**Review by mcoker — CHANGES_REQUESTED** [review]
Looks like there is a semantic token that fits the bill here https://github.com/patternfly/patternfly/blob/7f440dae1fb83b5ceb178c738a15425849b5503e/src/patternfly/base/tokens/tokens-default.scss#L209

The `--[100-900]` (or whatever the range is) tokens are what we refer to as "base" tokens (similar to [color palette vars](https://github.com/patternfly/patternfly/blob/v6/src/patternfly/base/tokens/tokens-palette.scss)) and in general we try to avoid using them unless there is no other/more appropriate semantic variable available.

**Review by mcoker — APPROVED** [review]
L🍕TM!!

**Review by kaylachumley — APPROVED** [review]
looks great! thanks!

**Pull request merged**